bundles/org.eclipse.core.runtime/src/org/eclipse/core/runtime/Platform.java

```diff
@@ -1462,4 +1464,47 @@ public static boolean inDebugMode() {
 	public static boolean inDevelopmentMode() {
 		return PlatformActivator.getContext().getProperty("osgi.dev") != null; //$NON-NLS-1$
 	}
+
+	/**
+	 * Retrieves the <b>native</b> system encoding ({@link Charset}) based on the
```

Comment:
The description is bad: i doubt there is something like THE native system encoding and they probably have nothing todo with system locale. 
It should help to add a description of the usecase where this charset can be used: https://docs.oracle.com/en/java/javase/18/docs/specs/man/java.html#java-command-line-argument-files
which calls it  "system default encoding". That wording is also used in https://docs.microsoft.com/en-us/windows/win32/api/winnls/nf-winnls-getsystemdefaultlocalename

Reply:
maybe even explicitly mention that it is NOT influenced by -Dfile.encoding and NOT meant for encoding files in general.

Reply:
> i doubt there is something like THE native system encoding and they probably have nothing todo with system locale.

No doubt, in the JVM is that THE encoding, and it **is** derived from system locale.

> NOT meant for encoding files in general.

For what else it is meant then?

> It should help to add a description of the usecase where this charset can be used

OK

Reply:
It can depend on the locale on windows, but on all other systems it is independent of locale.
The usecase is to get the system default locale because the receiver also uses exactly that (for example the java command line for Command-Line Argument Files).

Reply:
> all other systems it is independent of locale
Sure it is dependent on other systems too. On my RHEL 7 it depends for example on "LANG" environment. Set it to "C" and get US-ASCII, etc.

Reply:
mac has hardocded utf-8:
 https://github.com/openjdk/jdk/blob/master/src/java.base/unix/native/libjava/java_props_md.c#L453
independend from locale:
 https://github.com/openjdk/jdk/blob/master/src/java.base/unix/native/libjava/java_props_md.c#L104

I wonder if on your system we get what we need. Is the "current locale" also used for the "system default locale" in the java command line? and what then is the current locale of the java command line? Is it derived from the calling process?

Reply:
with "locale" you mean probably file encoding, and yes, it is derived from "LANG" on my system

Reply:
With locale i refer to https://wiki.debian.org/Locale / https://linux.die.net/man/3/setlocale. 
So if you set LANG in a JVM and then call the java command line from there is the modified locale respected?

Reply:
LANG is not a JVM system option, but environment variable. One does not set it in JVM, one gets it because one inherits the environment of the parent process. If one starts the child process with LANG overriden to a different value, the child JVM process uses this new LANG value to derive file encoding from it. So yes, if one explicitly changes environment variable LANG for the forked process,  forked process respects this change.

Reply:
user can set the environment variable in a launch...
![image](https://user-images.githubusercontent.com/103146188/173153981-728bf0b0-c87c-4ff0-8a18-bc827a9794a8.png)

it's not a usecase for me, but makes the use for launching theoretically unstable :-(

Reply:
Sure, every user can have different LANG value on same machine, and every process from same use can have that set to something else if user wants to do so. So for a process it doesn't matter who and where sets LANG, it can be system default or not, for the process it **is** the only "right" value.